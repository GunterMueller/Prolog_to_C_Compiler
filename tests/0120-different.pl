main :-
	VARS = [X, Y, Z],
	different(VARS),
	test(VARS).

test(VARS) :-
	VARS = [X, Y, Z],
	solve(X, Y, Z),
	write(VARS), nl, fail.
test(_).

solve(1, 1, 2).
solve(1, 2, 2).
solve(1, 2, 3).
solve(3, 2, 1).
solve(0, 2, 0).
solve(5, 6, 7).

%% not pairwise different, just adjacent ones
different([X]).
different([X|[Y|R]]) :-
	dif(X, Y),
	different([Y|R]).
