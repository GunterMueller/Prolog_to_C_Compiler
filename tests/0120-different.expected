[1,2,3]
[3,2,1]
[0,2,0]
[5,6,7]
